kornia/geometry/conversions.py

```diff
@@ -94,10 +95,9 @@ def convert_points_from_homogeneous(
     # set the results of division by zeror/near-zero to 1.0
     # follow the convention of opencv:
     # https://github.com/opencv/opencv/pull/14411/files
-    scale: torch.Tensor = torch.where(
-        torch.abs(z_vec) > eps,
-        torch.tensor(1.) / z_vec,
-        torch.ones_like(z_vec))
+    mask: torch.Tensor = torch.abs(z_vec) > eps
+    scale: torch.Tensor = torch.ones_like(z_vec).masked_scatter_(
+        mask, torch.tensor(1.0) / z_vec[mask])
 
     return scale * points[..., :-1]
```

Comment:
I think that this is still not safe if we have no control on `points` cause a scale safe value could be unsafe if `points` are quite extreme.

Reply:
You mean to return valid points, or the mask used?

Reply:
I mean you can guarantee that the scale is enough safe for a given precision but then it will be always responsibility to the user to constrain given arg `points`. The final precision is always  `scale * points` so if points could be anythings cause are in the learning process/pipeline you cannot guarantee to have enough precision.

Reply:
and what do you suggest?

Reply:
I think that without filtering Points with a clipping volume we cannot control points internally in the method. See http://www.cburch.com/cs/490/sched/02-15/clipping.html.
I we don't clip we need advise users that they need to self-control the input points (if they can).